sktm/jenkins.py

```diff
@@ -82,13 +86,46 @@ def __get_build(self, job, buildid):
                 return build
             except Exception as e:
                 logging.warning("catch %s: %s" % (type(e), e))
-                logging.info("now sleep 60s and try again")
-                time.sleep(60)
+                logging.info("now sleep %ds and try again" % interval)
+                time.sleep(interval)
 
         logging.error("fail to get build after retry %d times" %
                       self.retry_cnt)
         raise e
 
+    def __get_job_prop(self, job, method, interval):
+        """
+        Get property of Jenkins job. Retry Jenkins self.retry_cnt times
+        in case of temporary network failures.
+
+        Args:
+            job:        Jenkins job.
+            method:     Job method, it can be:
+                        - get_build_ids
+                        - get_last_build
+            interval:   Seconds to sleep before retrying.
+
+        Return:
+            job property if succeed, else raise the last exception.
+        """
+        for i in range(self.retry_cnt):
+            try:
+                return getattr(job, method)()
+            except Exception as e:
```

Comment:
In case invalid method name is passed, it would be nice to catch `AttributeError` and log and raise the exception directly, before catching the generic exception. Would save some time with debugging, what do you think?

Reply:
Agree. The related code is updated.